### DISCUSSION TITLE
Feature Request: encryption at rest

### DISCUSSION THREAD

**Discussion by LukeWheeldon** [oss-apps/split-pro#304] [Ideas]
Mirroring [this issue](https://github.com/spliit-app/spliit/issues/34) on Spliit.

Financial data is evidently very private, yet there has yet to be one expenses splitting app that is protecting the privacy of its users. Spliit seems to be working on it though, and I am curious about the thoughts of @KMKoushik on this matter before I start relying on this app myself.

**Comment by KMKoushik** [comment]
i'm all in for the privacy and wanted to see this happen, that being said i'm very much constrained with the time actually. 

if anyone will to work on this provide the proposal. 

main problems that i see is,

- we have groups and friends, if e2e encryption we they need to share the keys, which seems to be very tedious
- alternatively we can simply, encrypt at rest but that won't private enough.

if there's way where its seamless to incorprate this, i'll be very happy. these are my thoughts and i'm not expert in this.

**Comment by LukeWheeldon** [comment]
Thank you for your answer. I totally understand that it would most likely take time you do not have.

What I can say is that Proton is doing sharing of keys between users in a way that feels quite seamless and given it is open source, I am hopeful there is a way to reuse their work somehow.

Maybe worth doing like Spliit and keeping this issue open so that whenever someone with the right skills and motivation see it, it can potentially happen. Either way thank you for working on this great project!

**Comment by KMKoushik** [comment]
that make sense, i'm curious with the proton stuff can you share a link that i can look into?

**Comment by LukeWheeldon** [comment]
sure thing, this link has some interesting details about the inner workings of sharing data between users in an e2ee fashion: https://proton.me/blog/protoncalendar-security-model
